### PR TITLE
[Code Coverage] `test_profiler`

### PR DESCRIPTION
**Pull request by LukeLIN-web**
rename  `group_by(events, keyfn):` to  `_group_by(events, keyfn):`
rename `test_get_gpu_memory_from_gc():`
fix print function.
add `test_profiler`

### CODE REVIEW

**Comment by LukeLIN-web** [comment]
I thought that ` __str__` will be tested by `print`, but not , why?

File: test/profile/test_profiler.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    dataset = get_dataset(name='Cora')
+    use_cuda = torch.cuda.is_available()
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    data = dataset[0].to(device)
+    model = GraphSAGE(dataset.num_features, hidden_channels=64, num_layers=3,
+                      out_channels=dataset.num_classes).to(device)
+    model.train()
+    with Profiler(model, profile_memory=True, use_cuda=use_cuda) as prof:
+        model(data.x, data.edge_index)
+        print(prof)
```
Comment (lightaime) [inline test/profile/test_profiler.py:20]: Should we assert that memory usage is larger than zero? WDYT?

Comment (LukeLIN-web) [reply test/profile/test_profiler.py:19]: > Should we assert that memory usage is larger than zero? WDYT?

It seems that the profiler currently doesn't have any other output interfaces except `__str__` for print. Maybe I could add it. I refer    [Pytorch profiler](https://pytorch.org/tutorials/recipes/recipes/profiler_recipe.html). It also only has print API.

**Review by rusty1s — APPROVED** [review]
This module is sadly a bit unfinished, so it's quite hard to add meaningful tests to it. I'm sorry :( Still, this looks good. Thank you very much!

**Pull request merged**